tests/test_auth.py

```diff
@@ -0,0 +1,86 @@
+"""
+Test the authentication for each route.
+"""
+
+import types
+
+import jwt
+import pytest
+from fastapi import Depends, FastAPI
+from fastapi.testclient import TestClient
+
+# from pv_site_api.app import create_app
+from pv_site_api.auth import Auth
+
+# Use symetric algo for simplicity.
+ALGO = "HS256"
+SECRET = "my precious"
+
+
+@pytest.fixture
+def auth(config, monkeypatch):
+    """Our Auth object but with the Auth0 part monkey patched."""
+    auth = Auth(
+        api_audience=config.AUTH0_API_AUDIENCE,
+        domain=config.AUTH0_DOMAIN,
+        algorithm=ALGO,
+    )
+
+    class MockJwksClient:
+        def get_signing_key_from_jwt(self, token):
+            return types.SimpleNamespace(key=SECRET)
+
+    monkeypatch.setattr(auth, "_jwks_client", MockJwksClient())
+
+    yield auth
+
+
+@pytest.fixture
+def trivial_client(config, auth):
+    """A client with only one restricted route."""
+
+    app = FastAPI()
+
+    # Add a route that depends on authorization.
+    @app.get("/route", dependencies=[Depends(auth)])
+    def route():
+        pass
+
+    yield TestClient(app)
+
+
+def _make_header(token):
+    return {"Authorization": f"Bearer {token}"}
+
+
+def test_auth_happy_path(trivial_client, config):
```

Comment:
I wonder if we should split the test into `unittest` and `intergration`. The unittest could have no aith on it, and the intergations could have the real auth on them

Reply:
Actually ive just seen you have monkey patched it, so maybe its fine